### DISCUSSION TITLE
Allow setting column list for 'SET NULL' and 'SET DEFAULT' actions

### DISCUSSION THREAD

**Discussion by Kalos-S** [mikro-orm/mikro-orm#5259] [Ideas]
Implementing this feature into the PostgreSQL driver would be highly beneficial to serve use cases where a column is both a primary-key and a foreign-key. A clear explanation is given in the last example of section [5.4.5](https://www.postgresql.org/docs/15/ddl-constraints.html#DDL-CONSTRAINTS-FK) of the PostgreSQL documentation.

This could be implemented by either adding a new `deleteRuleColumns` option to decorators such as `@ManyToOne` or automatically excluding foreign-keys which also are primary-keys from the onDelete action.

**Comment by B4nan** [comment]
Can you summarize why is this needed? 

It looks like knex does not allow doing this, so supporting it here would mean yet another monkey patching, which I am not very keen to do. You should be able to do such changes in migrations and it shouldn't produce any schema diffs afterward, as this is not something we infer from the schema.

edit: I guess I find the important sentence in there

> Without the specification of the column, the foreign key would also set the column tenant_id to null, but that column is still required as part of the primary key.

I guess it would really make sense to do this automatically. I'll have a look at how hard this would be to patch into knex.
